In [10]:
import ujson
import gzip
import matplotlib
import pandas
import os
import re
import time
import pprint

In [43]:
def parse_type(type, num_examples = 5):
    examples = []
    num_found = 0
    with gzip.open('2015-10-01-15.json.gz') as f:
        for i, line in enumerate(f):
            json_data = ujson.loads(line)
            if (json_data['type'] == type):
                examples.append(json_data)
                num_found += 1
                if (num_examples < num_found):
                    f.close()
                    return(examples)
        f.close()
    return(examples)
def parse_notype(num_examples = 5):
    examples = []
    num_found = 0
    with gzip.open('2015-10-01-15.json.gz') as f:
        for i, line in enumerate(f):
            json_data = ujson.loads(line)
            if not json_data['type']:
                examples.append(json_data)
                num_found += 1
                if (num_examples < num_found):
                    f.close()
                    return(examples)
        f.close()
    return(examples)

def parse_language_event(num_examples = 5):
    examples = []
    num_found = 0
    regexp = re.compile(r'language')
    with gzip.open('2015-10-01-15.json.gz') as f:
        for i, line in enumerate(f):
            if regexp.search(line.decode()) is not None:
                json_data = ujson.loads(line)
                examples.append(json_data)
                num_found += 1
                if (num_examples < num_found):
                    f.close()
                    return(examples)
        f.close()
    return(examples)

In [48]:
language_events = []
for event in parse_language_event(9999999):
    language_events.append(ujson.dumps(event))
print("done")

done
['{"public":true,"payload":{"pull_request":{"assignee":{"gists_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/gists{\\/gist_id}","organizations_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/orgs","following_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/following{\\/other_user}","avatar_url":"https:\\/\\/avatars.githubusercontent.com\\/u\\/4551878?v=3","id":4551878,"subscriptions_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/subscriptions","type":"User","received_events_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/received_events","html_url":"https:\\/\\/github.com\\/wvu-r7","repos_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/repos","followers_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/followers","starred_url":"https:\\/\\/api.github.com\\/users\\/wvu-r7\\/starred{\\/owner}{\\/repo}","site_admin":false,"gravatar_id":"","login":"wvu-r7","url":"https:\\/\\/api.github.com\\/users\\/wvu-r7","events_url":"https:\\/\\/api.github.com\\/users\\

In [50]:

types = []
for i, line in enumerate(language_events):
    json_data = ujson.loads(line)
    types.append(json_data['type'])


In [51]:
pandas.DataFrame({"types":types}).groupby("types").size().order(ascending=False)

types
PullRequestEvent                 2531
ForkEvent                        1138
PullRequestReviewCommentEvent    1002
IssueCommentEvent                 100
WatchEvent                         67
PushEvent                          58
IssuesEvent                        29
CreateEvent                        23
ReleaseEvent                        2
dtype: int64

## example types

PushEvent                        18993
CreateEvent                       4944
IssueCommentEvent                 4299
WatchEvent                        2916
PullRequestEvent                  2531
IssuesEvent                       1846
ForkEvent                         1138
PullRequestReviewCommentEvent     1002
DeleteEvent                        924
GollumEvent                        346
MemberEvent                        330
CommitCommentEvent                 227
ReleaseEvent                       134
PublicEvent                         49

In [23]:
for event in parse_type("CreateEvent", 20):
    print("\n\n")
    print(ujson.dumps(event))
    print("\n\n")
    pprint.pprint(event)




{"public":true,"payload":{"ref_type":"repository","description":"done","master_branch":"master","pusher_type":"user","ref":null},"id":"3199013468","actor":{"login":"G3r4rd0","url":"https:\/\/api.github.com\/users\/G3r4rd0","avatar_url":"https:\/\/avatars.githubusercontent.com\/u\/14824205?","id":14824205,"gravatar_id":""},"created_at":"2015-10-01T15:00:00Z","repo":{"url":"https:\/\/api.github.com\/repos\/G3r4rd0\/p","name":"G3r4rd0\/p","id":43502925},"type":"CreateEvent"}



{'actor': {'avatar_url': 'https://avatars.githubusercontent.com/u/14824205?',
           'gravatar_id': '',
           'id': 14824205,
           'login': 'G3r4rd0',
           'url': 'https://api.github.com/users/G3r4rd0'},
 'created_at': '2015-10-01T15:00:00Z',
 'id': '3199013468',
 'payload': {'description': 'done',
             'master_branch': 'master',
             'pusher_type': 'user',
             'ref': None,
             'ref_type': 'repository'},
 'public': True,
 'repo': {'id': 43502925,
          '